### PR TITLE
Add section on subsetting

### PR DESCRIPTION
**Pull request by RiboRings**
Add section on subsetting as 2.6 in "Data Infrastructure". Change name of the parent section  to "Data manipulation"

### CODE REVIEW

File: 04_datainfra.Rmd [hunk 1]
```diff
@@ -389,7 +389,7 @@ features, but this is not a must have. This means there can be features, which
 are not linked to nodes, and nodes, which are not linked to features. To change
 the links in an existing object, the `changeTree` function is available.
 
-## Data conversion
+## Data manupulation
```
Comment (antagomir) [inline 04_datainfra.Rmd:392]: fix typo

File: 04_datainfra.Rmd [hunk 2]
```diff
@@ -409,6 +409,167 @@ molten_se <- meltAssay(se,
...
+
+#### Subset by sample (column-wise)
+
+For the sake of demonstration, here we will extract a subset containing only the samples of human origin (feces, skin or tongue), stored as `SampleType` within `colData(se)`.
+
+First, we would like to see all the possible values that `SampleType` can take on and how frequent those are: 
+
+```{r}
+# inspect possible values for SampleType
+unique(colData(se)$SampleType)
```
Comment (antagomir) [inline 04_datainfra.Rmd:448]: I think `unique(se$SampleType)` gives the same and is shorter. Check & change here and elsewhere

File: 04_datainfra.Rmd [hunk 3]
```diff
@@ -409,6 +409,167 @@ molten_se <- meltAssay(se,
...
+    kableExtra::scroll_box(width = "100%")
+```
+
+**Note**: after subsetting, expect the number of columns to equal the sum of the recurrences of the samples that you are interested in. For instance, `ncols = Feces + Skin + Tongue = 4 + 3 + 2 = 9`.
+
+Next, we _logical index_ across the columns of `se` (make sure to leave the first index empty to select all rows) and filter for the samples of human origin. For this, we use the information on the samples from the meta data `colData(se)`.
+
+```{r}
+# subset by sample
+se_subset_by_sample <- se[ , colData(se)$SampleType == "Feces" | colData(se)$SampleType == "Skin" | colData(se)$SampleType == "Tongue"]
```
Comment (antagomir) [inline 04_datainfra.Rmd:467]: `se$SampleType %in% ("Feces", "Skin", "Tongue")`

File: 04_datainfra.Rmd [hunk 4]
```diff
@@ -409,6 +409,167 @@ molten_se <- meltAssay(se,
...
+
+For subsetting by Kingdom, agglomeration does not apply, whereas for the other ranks it can be applied if necessary.
+
+#### Subset by sample and feature
+
+Finally, we can subset data by sample and feature at once. The resulting subset contains all the samples of human origin and all the features of Phyla "Actinobacteria" or "Chlamydiae".
+
+```{r}
+# subset by sample and feature and get rid of NAs
+se_subset_by_sample_feature <- se[(rowData(se)$Phylum == "Actinobacteria" | rowData(se)$Phylum == "Chlamydiae") & !is.na(rowData(se)$Phylum), colData(se)$SampleType == "Feces" | colData(se)$SampleType == "Skin" | colData(se)$SampleType == "Tongue"]
```
Comment (antagomir) [inline 04_datainfra.Rmd:564]: see above

File: 04_datainfra.Rmd [hunk 5]
```diff
@@ -409,6 +409,167 @@ molten_se <- meltAssay(se,
...
+# show dimensions
+dim(se_subset_by_sample_feature)
+```
+
+**Note**: the dimensions of `se_subset_by_sample_feature` agree with those of the previous subsets (9 columns filtered by sample and 1652 rows filtered by feature).
+
+If a study was to consider and quantify the presence of Actinobacteria as well as Chlamydiae in different sites of the human body, `se_subset_by_sample_feature` might be an ideal subset to start with.
```
Comment (antagomir) [inline 04_datainfra.Rmd:571]: replace "ideal" with "suitable"

**Review by antagomir — CHANGES_REQUESTED** [review]

**Comment by RiboRings** [comment]
Hey! I fixed the mistakes. Thanks for all the tips!

**Comment by antagomir** [comment]
I may move some of this to course materials later but this is good for now, thanks!

**Pull request merged**